### PR TITLE
Add error handling and UI feedback when submitting comment resulting in a 503 error

### LINKED ISSUE #14227: No UI Feedback when submitting comments that result in an HTTP 503 error.
**Describe the bug**

While investigating the cause of #14183, I discovered that we were not handling the case when the server returns an [HTTP 503](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/503) error code.

https://github.com/forem/forem/blob/e50db312fb5181c674a80b0c0e86f06752be8afc/app/assets/javascripts/initializers/initializeCommentsPage.js.erb#L293-L300

If the `response.status`  is `503`, the Promise `response.json()` on line 294 will fail resulting in the member of a Forem having no idea what happened when they submitted their comment.

image

**To Reproduce**

1. This can't be produced unless the server returns an HTTP 503 error, but you can see in #14183 how it happened.

**Expected behavior**

A member of a Forem submits a comment. If there is a server error when they submit the comment, they should receive some feedback that the comment submission failed.

**Screenshots**

**Desktop (please complete the following information):**

- OS, version: N/A
- Browser, version: N/A

**Smartphone (please complete the following information):**

- Device: N/A
- OS, version: N/A
- Browser, version: N/A

**Additional context**

### PR DESCRIPTION
**Pull request by zeotuan**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
HTTP 503 response status code is uncaught in promise when commenting cause comment to sometime to fail silently without any indicator 
## Related Tickets & Documents
Fixes #14227
## QA Instructions, Screenshots, Recordings
See #14227

_Please replace this line with instructions on how to test your changes, a note
on the devices and browsers this has been tested on, as well as any relevant
images for UI changes._

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://docs.forem.com/frontend/accessibility)._

## Added/updated tests?

- [ ] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [x] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://admin.forem.com/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] I've updated the README or added inline documentation
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: app/assets/javascripts/initializers/initializeCommentsPage.js.erb [hunk 1]
```diff
@@ -297,6 +297,13 @@ function handleCommentSubmit(event) {
           } else {
             showUserAlertModal('Error posting comment', 'Your comment could not be posted due to an error: ' + errorReponse.error, 'OK')
           }
+        }).catch(function parseError(error){
+            if(responseStatus === 503){
+              showUserAlertModal('Error posting comment', 'your comment could not be posted due to an error: HTTP 503 SERVER ERROR' , 'OK')
+            }else{
+              showUserAlertModal('Error posting comment', 'your comment could not be posted due to some unknown error', 'OK')
+            } 
+
```
Comment (rhymes) [inline app/assets/javascripts/initializers/initializeCommentsPage.js.erb:306]: It works well! A note:

I don't think we should tell the user, which can't be at all technical, that the error is a 503, but in general I don't think we should actually make a distinction between any of the 500 errors raised by `response.json()`. I don't think we need the `if` condition, we can just display:

```ruby
showUserAlertModal('Error posting comment', 'Your comment could not be posted due to a server error', 'OK')
```

This way we don't alarm the user but we tell them... something.

What do you think?

Comment (nickytonline) [reply app/assets/javascripts/initializers/initializeCommentsPage.js.erb:306]: What @rhymes said. 😎

Comment (zeotuan) [reply app/assets/javascripts/initializers/initializeCommentsPage.js.erb]: Got it! i have made the change.

**Comment by nickytonline** [comment]
This looks @zeotuan! It would be great to add an end to end test for this. We can use [cy.intercept](https://docs.cypress.io/api/commands/intercept#Usage) to simulate an HTTP 503 error.

You can add a test to the `commentOnArticle.spec.js` file. If you're unfamiliar with E2E testing and or [Cypress](https://www.cypress.io/), check out our [E2E testing documentation](https://docs.forem.com/tests/e2e-tests/).

If you need any assistance with this, do not hesitate to ask.

**Comment by zeotuan** [comment]
I have just added a test for the new change.

**Review by rhymes — APPROVED** [review]
Awesome work @zeotuan, thank you!

**Review by aitchiss — APPROVED** [review]
Nice one! Great having the cypress test for this ✨

**Pull request merged**